rust/lance/src/dataset.rs

```diff
@@ -3039,6 +3089,53 @@ mod tests {
         assert!(fragments[0].metadata.deletion_file.is_some());
     }
 
+    #[rstest]
+    #[tokio::test]
+    async fn test_table_metadata(
+        #[values(LanceFileVersion::Legacy, LanceFileVersion::Stable)]
+        data_storage_version: LanceFileVersion,
```

Comment:
It's unclear to me that it's worth parameterizing over the data storage version. It's just something that's written to the manifest, so the kind of data files should be orthogonal.

Reply:
Easy done - I had just copied and pasted this from another test.

Reply:
Done.